src/gpu_sysman.c

```diff
@@ -59,7 +59,7 @@
 #include "utils/common/common.h"
 
 #define PLUGIN_NAME "gpu_sysman"
-#define METRIC_PREFIX "collectd_" PLUGIN_NAME "_"
+#define METRIC_PREFIX "sysman.gpu."
```

Comment:
Sounds reasonably, but have you suggestion for differentiating GPU plugins metrics from from each other? Adding vendor labels to all metrics would be very wasteful...

How / could I add vendor info as resource (i.e. output only once per query, not per-metric)?

Reply:
Thanks!

I guess I need to generate per-GPU label sets at plugin init time, add them to families in the common submit function, and reset those label sets only when plugin is uninitialized.

Btw. There are already few per-GPU labels added to all metric families in the submit function (`pci_dev`, `pci_dev`, `dev_file`) which I can move over to resource labels, see: https://github.com/collectd/collectd/blob/collectd-6.0/src/gpu_sysman.c#L770

Reply:
@octo Storing per-GPU info in per-metric-family resource, instead of metric labels, breaks metrics reporting.

When multiple GPUs are reporting (similar) metric values, so that only difference is in their resource label values, `write_prometheus` will report metrics and target_info only for the _first_ GPU, and _ignore metrics submitted for all the other devices_.

PS. logging metrics directly from the plugin itself was really useful for debugging this.

Reply:
If I add unique per-GPU label (back) to metrics, then  `write_prometheus` (again) reports metrics for all GPUs.

However, it still reports `target_info` only for the first GPU.  I.e. resource handling is not really useful in its current form. :-/

Reply:
With #4284, and the commit adding GPU ID as "service.instance.id" this started working OK.   I.e. I can next add rest of the labels suggested by the spec.